app/pages/MainPage/Main.js

```diff
@@ -54,66 +49,47 @@ class Main extends React.Component {
       dataSource: new ListView.DataSource({
         rowHasChanged: (row1, row2) => row1 !== row2
       }),
-      typeIds: [],
       typeList: {}
     };
   }
 
   componentDidMount() {
-    const { readActions } = this.props;
-    DeviceEventEmitter.addListener('changeCategory', (typeIds) => {
-      typeIds.forEach((typeId) => {
-        readActions.requestArticleList(false, true, typeId);
-        pages.push(1);
-      });
+    this.retrieveArticleLists(this.props.selectedCategoryIds);
+
+    store.get('typeList').then((typeList) => {
       this.setState({
-        typeIds
-      });
-    });
-    InteractionManager.runAfterInteractions(() => {
-      store.get('typeIds').then((typeIds) => {
-        if (!typeIds) {
-          return;
-        }
-        typeIds.forEach((typeId) => {
-          readActions.requestArticleList(false, true, typeId);
-          pages.push(1);
-        });
-        store.get('typeList').then(typeList =>
-          this.setState({
-            typeIds,
-            typeList
-          })
-        );
+        typeList
       });
     });
   }
 
   componentWillReceiveProps(nextProps) {
     const { read } = this.props;
+    this.retrieveArticleListsWhenNeeded(
```

Comment:
since I removed event listeners, so **every time** when category button is pressed, the method will be called.So the idea is just to retrieve the article lists of new selected categories compared to the previously selected categories. The thing is I can't find a better way to only retrieve article lists when the user is done selecting them. would love to hear your suggestions. :)

Reply:
sure,I will try migrating listview to flatlist.

Reply:
Wonderful！